framework/include/utils/InputParameters.h

```diff
@@ -313,6 +339,52 @@ class InputParameters : public libMesh::Parameters
                            const std::string & syntax,
                            const T & value,
                            const std::string & doc_string);
+  template <typename T>
+  void addCommandLineParam(const std::string & name,
+                           const std::string & syntax,
+                           const std::initializer_list<typename T::value_type> & value,
+                           const std::string & doc_string)
+  {
+    addCommandLineParam<T>(name, syntax, T{value}, doc_string);
+  }
+
+  /**
+   * Add a command line parameter with an optional value.
+   *
+   * This is a deprecated option and only remains for two parameters:
+   * "mesh_only" and "recover". There are issues with command line
+   * parameters with optional values because if a value following
+   * one of these is a hit cli parameter, we don't know if we should
+   * apply it to the optional option or as a hit parameter.
+   *
+   * @param name The name of the parameer
+   * @param syntax Space separated list of command-line switch syntax that can set this option
+   * @param value The default value to assign
+   * @param doc_string Documentation.  This will be shown for --help
+   */
+  ///@{
+  template <typename T>
+  void addOptionalValuedCommandLineParam(const std::string & name,
+                                         const std::string & syntax,
+                                         const T & value,
+                                         const std::string & doc_string);
+  template <typename T>
+  void
+  addOptionalValuedCommandLineParam(const std::string & name,
+                                    const std::string & syntax,
+                                    const std::initializer_list<typename T::value_type> & value,
+                                    const std::string & doc_string)
+  {
+    addOptionalValuedCommandLineParam<T>(name, syntax, T{value}, doc_string);
+  }
```

Comment:
Looks like T should be an instantiation of `std::optional`, but below you call ` addOptionalValuedCommandLineParam<std::string>`. What's going on here?!

Reply:
I'm not sure I follow. This allows you to do a

```
addOptionalValueCommandLineParam<std::string>("name", "--name", {}, "doc");
```

Not sure it's really needed though as this is only used three times and cannot be used for any new params

Reply:
`std::initializer_list<typename T::value_type> & value` resolves to `std::initializer_list<std::string:value_type>` which is `std::initializer_list<CharT>` .

Yeah, this compiles, and doesn't produce nonsense for an empty `{}` initializer list... but it's hella funky.